tasks/drupal.xml

```diff
@@ -21,142 +21,68 @@
-    <target name="drupal-prepare-filesystem">
-        <fail unless="drupal.root" />
-        <fail unless="drupal.sites_subdir" />
-        <fail unless="drupal.settings.file_public_path" />
-
-        <!-- Create the Drupal modules, themes, profiles, and sites directories. -->
-        <foreach target="create-placeholder" param="placeholder_for">
-            <filelist dir="${build.dir}" files="${drupal.root}/modules/custom,${drupal.root}/themes/custom,${drupal.root}/profiles/custom,${drupal.root}/sites/${drupal.sites_subdir}" />
-        </foreach>
-
-        <!-- The site directory needs to have some perms. -->
-        <chmod file="${drupal.root}/sites/${drupal.sites_subdir}" mode="750" />
-
-        <!-- The public files directory, and everything in it, needs to be world writable. -->
-        <resolvepath propertyName="drupal.settings.file_public_path.resolved" file="${drupal.root}/${drupal.settings.file_public_path}" />
-        <exec command="chmod -R 777 ${drupal.settings.file_public_path.resolved}" />
-    </target>
-
-
-    <!-- Target: drupal-prepare-settings -->
-    <target name="drupal-prepare-settings">
-        <fail unless="build.drupal.settings" />
-        <fail unless="build.drupal.settings_dest" />
-
-        <copy file="${build.dir}/${build.drupal.settings}" tofile="${build.dir}/${build.drupal.settings_dest}" overwrite="true" mode="555">
-            <filterchain>
-                <expandproperties />
-            </filterchain>
-        </copy>
-    </target>
-
-
-    <!-- Target: drupal-prepare-services -->
-    <target name="drupal-prepare-services">
-        <fail unless="build.drupal.services" />
-        <fail unless="build.drupal.services_dest" />
-
-        <copy file="${build.dir}/${build.drupal.services}" tofile="${build.dir}/${build.drupal.services_dest}" overwrite="true" mode="644">
-            <filterchain>
-                <expandproperties />
-            </filterchain>
-        </copy>
-    </target>
+    <!--
+        Target: drupal-validate-clean-config
 
+        Validate that the Drupal config directory doesn't have local modifications,
+        to ensure consistent installation.
+    -->
+    <target name="drupal-validate-clean-config" description="Validate clean drupal config directory.">
+      <!-- Assembling this property at run time. -->
+      <property name="drupal.validate_clean_config.dir" value="${build.dir}/${drupal.root}/${drupal.config_sync_directory}" />
 
-    <!-- Target: create-placeholder -->
-    <target name="create-placeholder">
-        <fail unless="placeholder_for" message="The 'placeholder_for' property is required." />
-        <resolvepath propertyName="placeholder_for.resolved" file="${placeholder_for}" dir="${build.dir}" />
+        <!-- Use git to list local (not yet committed) changes to the config directory. -->
+        <exec command="git status --porcelain ${drupal.validate_clean_config.dir}" outputProperty="modified_files" />
```

Comment:
I encountered a strange situation where, when the `config` directory doesn't exist, the build fails here.  However, using `bypass` doesn't display the `modified_files`, implying that it is indeed empty.  So, I would expect that if it fails here, it should print the message "Continuing install with these local changes..." when using `bypass`.

I only ran into this because I didn't install Drupal as part of `the-build-installer`.  I was following along with the Drupal Skeleton readme, which instructs running the installer outside the VM before `vagrant up`.  Do you think we should change that when this is merged, or is there a way we can make `phing install` do the initial install if it has not been already?

Reply:
Oh, that's an interesting idea -- to offer to run the first-install target if Drupal hasn't been installed yet. I've been wondering how to handle this, since I don't necessarily expect folks to install the-build from within a Vagrant environment...

Reply:
Ok, I've tried to address this issue! I've made it so that, if there's no config to install from, the-build offers to install a fresh Drupal when you run `phing install`. This didn't involve changing anything else about the `drupal-validate-clean-config` target, but let me know if that part still gets in your way.